pkg/operator/resource_reconciler.go

```diff
@@ -357,12 +381,43 @@ func (rr *ResourceReconciler) onStatefulSetUpdate(old, cur *appsv1.StatefulSet)
 	rr.logger.Debug("StatefulSet updated")
 	rr.metrics.TriggerByCounter("StatefulSet", UpdateEvent).Inc()
 
-	if !rr.hasStateChanged(old, cur) {
+	// TODO: Uncomment this when Prometheus Agent DaemonSet's status has been supported.
+	/*if !rr.hasStateChanged(old, cur) {
 		// If the statefulset state (spec, labels or annotations) hasn't
 		// changed, the operator can only update the status subresource instead
 		// of doing a full reconciliation.
 		rr.EnqueueForStatus(obj)
 		return
+	}*/
+
+	rr.EnqueueForReconciliation(obj)
+}
+
+func (rr *ResourceReconciler) onDaemonSetUpdate(old, cur *appsv1.DaemonSet) {
+	rr.logger.Debug("update handler", "resource", "daemonset", "old", old.ResourceVersion, "cur", cur.ResourceVersion)
+
+	if rr.DeletionInProgress(cur) {
+		return
+	}
+
+	if !rr.hasObjectChanged(old, cur) {
+		return
+	}
+
+	obj := rr.syncer.Resolve(cur)
+	if obj == nil {
+		return
+	}
+
+	rr.logger.Debug("DaemonSet updated")
+	rr.metrics.TriggerByCounter("DaemonSet", UpdateEvent).Inc()
+
+	if !rr.hasStateChanged(old, cur) {
+		// If the daemonset state (spec, labels or annotations) hasn't
+		// changed, the operator can only update the status subresource instead
+		// of doing a full reconciliation.
+		rr.EnqueueForStatus(obj)
```

Comment:
this is where we'd need a TODO?

Reply:
I'm unresolving this comment because I'm also a bit confused here. 

I think we still want to keep the `if !rr.hasStateChanged(old,cur) ` here, and only comment `rr.EnqueueForStatus(obj)` because there's an early return inside the if.

In other words, if the old and new daemonset are fundamentally equal, we'd like to avoid calling `rr.EnqueueForReconciliation` below.

Reply:
Yeah I got it now. Will uncomment the return part